src/aspect.ts

```diff
@@ -1,27 +1,80 @@
 import { Handle } from './interfaces';
 import { createHandle } from './lang';
 
+/**
+ * An object that provides the necessary APIs to be MapLike
+ */
+export interface MapLike<K, V> {
+	get(key: K): V;
+	set(key: K, value?: V): this;
+}
+
+/**
+ * An internal type guard that determines if an value is MapLike or not
+ *
+ * @param value The value to guard against
+ */
+function isMapLike(value: any): value is MapLike<any, any> {
+	return value && typeof value.get === 'function' && typeof value.set === 'function';
+}
+
+export interface Indexable {
```

Comment:
I've been pulled me up for the same before in `dojo/loader`. I believe it's the interface names more than anything, ie. `MapLike` above `Indexable`, that's not the correct order.
We either need to code to our guidelines or update them.